examples/restrictable-expression.flix

```diff
@@ -0,0 +1,62 @@
+restrictable enum Expr[s] {
+    case Cst(Bool)
+    case Var(Int32)
+    case Not(Expr[s])
+    case And(Expr[s], Expr[s])
+    case Or(Expr[s], Expr[s])
+    case Xor(Expr[s], Expr[s])
+}
+
+def size(e: Expr[s]): Int32 = choose e {
+    case Expr.Cst(_) => 1
+    case Expr.Var(_) => 1
+    case Expr.Not(x) => size(x) + 1
+    case Expr.And(x, y) => size(x) + size(y) + 1
+    case Expr.Or(x, y) => size(x) + size(y) + 1
+    case Expr.Xor(x, y) => size(x) + size(y) + 1
+}
+
+def eq(e1: Expr[s1], e2: Expr[s2]): Bool = ??? // nested choose
+
+def simplify(e: Expr[s]): Expr[(s -- <Expr.Xor>) ++ <Expr.Not>] =
+    choose* e {
+        case Expr.Var(x) => Expr.Var(x)
+        case Expr.Cst(b) => Expr.Cst(b)
+        case Expr.Not(x) => open Expr.Not(openExpr(simplify(x)))
+        case Expr.And(x, y) => Expr.And(???, ???)
```

Comment:
This depends on the other open?

Reply:
choose* still has the check that the body must be a tag, so i cant use openExpr

Reply:
OK, but that can be lifted, yes?

Reply:
We've found that I can't, but i think that'll be discussed elsewhere